### PR TITLE
Dense vae pmm

### PR DESCRIPTION
**Pull request by maffettone**
1. Apply black. Double check you are using the dev instructions and have pre-commit hooks set up in your repo. 
2. Made static VAE methods static. 
3. Build a __call__ method to super the Model call. This is pretty experimental on my end, but should work and provide a bit of convenience with turning things like Training (backprop) on and off elegantly. When we do things like validation, or deployment, we want a way to call `vae.encode(x, training=False)`
4. First pass at a training protocol for the VAE. 
5. Lastly some docs and adjustment to data classes. It looks like the design here depends on a sigmoid output and crossentropy loss instead of MSE loss. That means our input data needs to be on [0,1]. This should be fine; however, the default preprocessing was to cast the data on [-1,1]. The tfrecords and datasets are stored as [0,1] though, so no sweat.

### CODE REVIEW

**Comment by maffettone** [comment]
Let me know if you have any questions, or want to walk through some of the code. Try and give it a run in the morning with the BTO tfrecords we have previously made/saved.

This is all presently untested, so it may take some debugging to polish off.
